services/src/pro/quota/mod.rs

```diff
@@ -0,0 +1,161 @@
+use futures::{
+    channel::mpsc::{channel, Receiver, Sender},
+    StreamExt,
+};
+use geoengine_operators::pro::meta::quota::{ComputationContext, ComputationUnit, QuotaTracking};
+use std::sync::Arc;
+
+use geoengine_datatypes::util::test::TestDefault;
+
+use crate::pro::users::UserId;
+
+use super::users::{UserDb, UserSession};
+
+// TODO: find a good default, choose w.r.t. resources or make configurable
+const QUOTA_CHANNEL_SIZE: usize = 1000;
+
+#[derive(Debug, Clone)]
+pub struct QuotaTrackingFactory {
+    quota_sender: Sender<ComputationUnit>,
+}
+
+impl QuotaTrackingFactory {
+    pub fn new(quota_sender: Sender<ComputationUnit>) -> Self {
+        Self { quota_sender }
+    }
+
+    pub fn create_quota_tracking(
+        &self,
+        session: &UserSession,
+        context: ComputationContext,
+    ) -> QuotaTracking {
+        QuotaTracking::new(
+            self.quota_sender.clone(),
+            ComputationUnit {
+                issuer: session.user.id.0,
+                context,
+            },
+        )
+    }
+}
+
+impl TestDefault for QuotaTrackingFactory {
+    fn test_default() -> Self {
+        let (quota_sender, _quota_receiver) = channel(QUOTA_CHANNEL_SIZE);
+        Self::new(quota_sender)
+    }
+}
+
+pub struct QuotaManager<U: UserDb + 'static> {
+    user_db: Arc<U>,
+    quota_receiver: Receiver<ComputationUnit>,
+}
+
+impl<U: UserDb + 'static> QuotaManager<U> {
+    pub fn new(user_db: Arc<U>, quota_receiver: Receiver<ComputationUnit>) -> Self {
+        Self {
+            user_db,
+            quota_receiver,
+        }
+    }
+
+    pub fn run(mut self) {
+        crate::util::spawn(async move {
+            while let Some(computation) = self.quota_receiver.next().await {
+                // TODO: issue a tracing event instead?
```

Comment:
Yeah, this will definitely blow our log.

Reply:
right now we log all the tracing events as well. The todo was meant on a semantic level.

Reply:
56ad2af66b5e7c59ad180b6b15f2405e6fc39e73